llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -2668,8 +2668,8 @@ SDValue DAGCombiner::visitADDLike(SDNode *N) {
     return C;
 
   // canonicalize constant to RHS
-  if (DAG.isConstantIntBuildVectorOrConstantInt(N0) &&
-      !DAG.isConstantIntBuildVectorOrConstantInt(N1))
+  if (DAG.isConstantIntBuildVectorOrConstantInt(peekThroughBitcasts(N0)) &&
```

Comment:
Can we do it in `isConstantIntBuildVectorOrConstantInt` instead?

Reply:
Yes, although we would need to make it return bool instead of the SDNode (which is barely used tbh).